src/bin/inx-chronicle/cli/analytics.rs

```diff
@@ -7,42 +7,46 @@ use chronicle::{
         influxdb::{AnalyticsChoice, InfluxDb},
         MongoDb,
     },
-    tangle::Tangle,
+    tangle::{InputSource, Tangle},
     types::tangle::MilestoneIndex,
 };
 use futures::TryStreamExt;
 
-pub async fn fill_analytics(
-    db: &MongoDb,
-    influx_db: &InfluxDb,
+pub async fn fill_analytics<I: InputSource + Clone + Send + Sync + 'static>(
+    db: MongoDb,
+    influx_db: InfluxDb,
     start_milestone: MilestoneIndex,
     end_milestone: MilestoneIndex,
     num_tasks: usize,
-    analytics: &[AnalyticsChoice],
+    analytics: Vec<AnalyticsChoice>,
```

Comment:
But why

Reply:
sorry ... I had some lifetime issues ... and this was an intermediary step.